### ISSUE TITLE
Add GUI Builder UI

### ISSUE THREAD

**Issue by adamfowleruk** [HaikuArchives/Paladin#269]
Drag and drop widgets to visually place. Auto layout support (like XCode)
@diversys

**Comment by diversys** [comment]
Maybe we can integrate one of already existing GUI builders https://github.com/HaikuArchives?q=gui

**Comment by adamfowleruk** [comment]
Had a look at BeBuilder. Fixed its compilation problems and issued a PR: https://github.com/adamfowleruk/BeBuilder/tree/feature-2019build
Looks like a good base, although buggy. Interesting concepts there.

**Comment by adamfowleruk** [comment]
Yeah the old Paladin PDesigner code base is in a terrible state. BeBuilder is about the best so far, but very basic and not at all user friendly (I think coding by hand may be easier...). Good starting code base though. Not sure about 'code generation' - I think that's the wrong approach. Prevents round trip re-engineering. I'll have a think.

**Comment by adamfowleruk** [comment]
Links for Auckland Layout Manager (ALM) style layout editor on Haiku from PulkoMandy:-
- https://www.haiku-os.org/blog/czeidler/2012-09-03_ale_auckland_layout_editor/ 
- https://www.researchgate.net/publication/262402078_The_Auckland_Layout_Editor_An_improved_GUI_layout_specification_process 
- https://github.com/czeidler/ALEditor

**Comment by adamfowleruk** [comment]
Also MeTOS IDE from Bruno:-

https://cvincent.pagesperso-orange.fr/logiciels/metos.html

**Comment by diversys** [comment]
Would be nice to make it build on Haiku since the project is pretty big:
https://github.com/HaikuArchives/MeTOS

**Comment by bitigchi** [comment]
Rather than a drag-drop builder, something that would immediately preview the UI immediately or on trigger would be neater, and it could easily applicable with the layout builder code.

For reference, SwiftUI works this way, and it is much better than dealing with storyboards. Storyboards are a pain in the butt to work with, especially with constraints.
